primitives/keystore/src/lib.rs

```diff
@@ -53,6 +53,14 @@ pub enum Error {
 pub trait CryptoStore: Send + Sync {
 	/// Returns all sr25519 public keys for the given key type.
 	async fn sr25519_public_keys(&self, id: KeyTypeId) -> Vec<sr25519::Public>;
+	/// Returns an sr25519 key pair given the public key, if available.
+	///
+	/// Note that if a key is available in the keystore, its private key may still not be retrivable.
+	async fn sr25519_key_pair(
```

Comment:
Why were those methods introduced? instead of adding them to the trait, we could just add them to `LocalKeystore`, no? Adding them to the trait means they're accessible and part of the interface for the remote signer as well.

Reply:
I just noticed the comment: https://github.com/paritytech/substrate/pull/7489#issuecomment-722359361 but doesn't explain why not limit the methods to LocalKeystore... can you please elaborate on the decision?

Reply:
I'm honestly fine either way. @bkchr or @gnunicorn would you mind to make a call on this?

Reply:
For reference, I don't think we have issues supporting this on remote keystores at all. In the worst case, it can always just return `Error::Unavailable`.

Reply:
I'm also not really convinced by this.

If @gnunicorn wants it, fine, but it also defeats the purpose of the remote keystore.

We could also merge this without the feature for now and come up with a solution when we need it. The key could for example also just be passed to the POW system without even using the keystore.

Reply:
@bkchr @rakanalh I see. I reverted the commit that introduces trait-level keypair functions!